pkg/cover/report.go

```diff
@@ -123,6 +129,20 @@ func (rg *ReportGenerator) prepareFileMap(progs []Prog) (map[string]*file, error
 	if !matchedPC {
 		return nil, fmt.Errorf("coverage doesn't match any coverage callbacks")
 	}
+	// Check that all collected coverage points have matching coverage callbacks.
+	// This check is not performed for targets that explicitly list "gcc" as KernelCompiler,
+	// because code coverage instrumentation does not work correctly in GCC < 14. For Clang
+	// and/or targets that do not set the compiler (e.g. in tests) the check is still done.
+	// See https://github.com/google/syzkaller/issues/4447 for more information.
+	if (len(allProgPCs) != len(matchedProgPCs)) && (rg.target.KernelCompiler != "gcc") {
+		for mismatch := range allProgPCs {
```

Comment:
This loop and following if are not needed.

Reply:
Sorry about that, you are right (and you've pointed at this in another patch already)